lib/luminork-server/src/search/query.rs

```diff
@@ -0,0 +1,317 @@
+        // TODO These partial ones should eventually match!
+        assert!(!SearchTerm::Match("nul".to_string()).match_null());
+        assert!(!SearchTerm::Match("nu".to_string()).match_null());
+        assert!(!SearchTerm::Match("n".to_string()).match_null());
+
+        assert!(!SearchTerm::Match("nulll".to_string()).match_null());
+        assert!(!SearchTerm::Match("nnull".to_string()).match_null());
+        assert!(!SearchTerm::Match("true".to_string()).match_null());
+        assert!(!SearchTerm::Match("false".to_string()).match_null());
+        assert!(!SearchTerm::Match("0".to_string()).match_null());
+        assert!(!SearchTerm::Match("1".to_string()).match_null());
+    }
+
+    #[test]
+    fn match_bool() {
+        assert!(SearchTerm::Match("true".to_string()).match_bool(true));
+        assert!(SearchTerm::Match("TRUE".to_string()).match_bool(true));
+        // TODO These partial ones should eventually match!
+        assert!(!SearchTerm::Match("tru".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("tr".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("t".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("truee".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("ttrue".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("true".to_string()).match_bool(false));
+
+        assert!(SearchTerm::Match("false".to_string()).match_bool(false));
+        assert!(SearchTerm::Match("FALSE".to_string()).match_bool(false));
+        // TODO These partial ones should eventually match!
+        assert!(!SearchTerm::Match("fals".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("fal".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("fa".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("f".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("falsee".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("ffalse".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("false".to_string()).match_bool(true));
+
+        assert!(!SearchTerm::Match("null".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("null".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("0".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("0".to_string()).match_bool(false));
+        assert!(!SearchTerm::Match("1".to_string()).match_bool(true));
+        assert!(!SearchTerm::Match("1".to_string()).match_bool(false));
+    }
+
+    #[test]
+    fn match_number() {
+        assert!(SearchTerm::Match("0".to_string()).match_number(&0.into()));
+        // assert!(SearchTerm::Match("-0".to_string()).match_number(&0.into()));
+
+        assert!(SearchTerm::Match("1".to_string()).match_number(&1.into()));
+        // assert!(SearchTerm::Match("1.0".to_string()).match_number(&1.into()));
```

Comment:
serde_json does some crazy stuff with numbers--it doesn't consider 1 == 1.0. (1.00 == 1.0 though, and 1 == 1). I'm opting not to fix this for now, but it needs love.